src/main/java/org/radarcns/management/web/rest/SubjectResource.java

```diff
@@ -300,10 +301,17 @@
             throws NotAuthorizedException {
         log.debug("REST request to get Subject : {}", login);
         Subject subject = subjectService.findOneByLogin(login);
+        Project project = subject.getActiveProject()
+                .flatMap(p ->  projectRepository.findOneWithEagerRelationships(p.getId()))
+                .orElse(null);
+
+        String projectName = project == null ? null : project.getProjectName();
```

Comment:
should we add InvalidStateException if the projectName is null?

Reply:
Is a project required for all subjects? If so, yes, it should be an invalid state. Otherwise, I don't see why a subject without a project should not be returned.

Reply:
I think the subject can be returned without a project. If the subject is `discontinued`, I think then the active project becomes `null`. I was more concerned if the `checkPermissionOnSubject` can handle null for project. If so, it looks good

Reply:
I've added a some tests for this, indeed, `null` project authorization behaves as expected.